offer-manager-app/src/main/groovy/life/qbic/portal/offermanager/components/product/MaintainProductsController.groovy

```diff
@@ -68,6 +66,27 @@ class MaintainProductsController {
         }
     }
 
+    /**
+     * Triggers the copy Use Case of a product
+     *
+     * @param category The products category which determines what kind of product is created
+     * @param description The description of the product
+     * @param name The name of the product
+     * @param unitPrice The unit price of the product
+     * @param unit The unit in which the product is measured
+     */
+    void copyProduct(ProductCategory category, String description, String name, double unitPrice, ProductUnit unit, ProductId productId){
+        try{
+            println("You did it!")
+            //ToDo how should the Id be provided to the Use Case?
```

Comment:
The ID is provided by the database when the new product is stored

Reply:
Yes for the new product that is correct. But we don't transfer the old productId, so how is this compared in the database?

Reply:
The old product id is stored in the copied product

Reply:
But I agree that this is not declarative enough. I also discussed with tobi that we should change something about the ids.. But this is not relevant now I think 🤔

Reply:
we could comment that the id is extracted from the product..just to indicate that for the future

Reply:
I added it to the method documentation thanks for clarifying